cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -49,24 +61,32 @@ protected Transformation getTransformation(String pluginType, String stageName)
     return getInitializedTransformation(stageName);
   }
 
+
+  public TransformExecutor<T> create(PipelinePhase pipeline) throws Exception {
+    return create(pipeline, Collections.<String>emptySet());
+  }
+
   /**
    * Create a transform executor for the specified pipeline. Will instantiate and initialize all sources,
    * transforms, and sinks in the pipeline.
    *
    * @param pipeline the pipeline to create a transform executor for
+   * @param pluginTypesToExclude which plugin types are not transform
    * @return executor for the pipeline
    * @throws InstantiationException if there was an error instantiating a plugin
    * @throws Exception if there was an error initializing a plugin
    */
-  public TransformExecutor<T> create(PipelinePhase pipeline) throws Exception {
-
+  public TransformExecutor<T> create(PipelinePhase pipeline, Set<String> pluginTypesToExclude) throws Exception {
```

Comment:
In SparkTransformExecutorFactory, we can override `getTransformation(String pluginType, String stageName)` rather than adding another argument here.